panel/models/card.ts

```diff
@@ -107,10 +107,10 @@ export class CardView extends ColumnView {
     }
 
     for (const child_view of this.child_views.slice(1)) {
+      child_view.model.visible = child_view.model.visible && !this.model.collapsed
```

Comment:
Hmm, I guess this is still problematic, because once visible=False it won't be restored to true subsequently.